src/sc/contentrules/localrole/configure.zcml

```diff
@@ -45,4 +55,18 @@
         factory=".action.LocalRoleAction"
         />
 
+    <plone:behavior
+        title="Principle from context"
+        description="Pick principle field from context item."
+        provides=".interfaces.IPrincipleFromContext"
+        for="plone.dexterity.interfaces.IDexterityContent"
+        />
+
+    <adapter
+        for=".interfaces.IPrincipleFromContext"
```

Comment:
If for some reason you still find out better to don't use the behavior (and create the field by hand TTW in Dexterity) we should change this:
```suggestion
        for="*"
```

Reply:
Jon makes some tests during a call with me and we realize that the StringInterpolation don't work like this. We need to create [one adapter for each field](https://github.com/plone/plone.stringinterp/blob/master/plone/stringinterp/configure.zcml#L27-L319).
Also I give the idea of an adapter to keep the right name of the field used in the new adapter.

Reply:
Another option is to change the way StringInterpolation works.

Reply:
forget about stringinterpolation. You are using something that is not well thought out and designed not for the user but over smart plone developers. Just have a list of fields and pick one and use it. Simple.

Reply:
The list of fields will be selected into the content rule, right?
So it will be a new field there, but I can't introspect at this point the content type.
Is the list of fields hardcoded?

Sorry Dylan, I can't understand how it will work.

Reply:
a list of all fields on all types. If the final field can't be found on the item then it does noting. The content rule might not be type specific in any case.